cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -107,6 +107,38 @@ csrfToken = hmac + "." + randomValue // Add the `randomValue` to the HMAC hash t
 response.setCookie("csrf_token=" + csrfToken + "; Secure") // Set Cookie without HttpOnly flag
 ```
 
+And the following code demonstrates validation of the CSRF token that is sent back from the client:
+
+```code
+// Get the CSRF token from the request
+csrfToken = request.getParameter("csrf_token") // From form field or header
+
+// Split the token to get the randomValue
+const tokenParts = csrfToken.split(".");
+const hmacFromRequest = tokenParts[0];
+const randomValue = tokenParts[1];
+
+// Recreate the HMAC with the current session and the randomValue from the request
+secret = readEnvironmentVariable("CSRF_SECRET") // HMAC secret key
```

Comment:
While passing secrets as environment variables is less bad than certain other scenarios, it is still not truly secure. E.g., see:

- https://www.trendmicro.com/en_us/research/22/h/analyzing-hidden-danger-of-environment-variables-for-keeping-secrets.html
- https://blog.diogomonica.com/2017/03/27/why-you-shouldnt-use-env-variables-for-secret-data/

We generally to not allow this except as a last resort. Rather, retrieve the secret CSRF_SECRET in something like AWS KMS or HashiCorp Vault, etc. (If we change this here, a corresponding change is needed on line 97 as well.)

Reply:
This is what the pseudocode in the existing section says. I kept it consistent.

Complicating pseudocode with details of proprietary services like AWS & HashiCorp doesn't seem like it will age well.

Reply:
Maybe just `getSecret("CSRF_SECRET")` ?

Reply:
> Maybe just `getSecret("CSRF_SECRET")` ? Environment variable, secret, or otherwise (future proof).

Make it `getSecretSecurely` or similar, and I could stand behind that.